account_vat_on_payment/account_bank_statement.py

```diff
@@ -0,0 +1,289 @@
+        return vals
+
+    def _prepare_shadow_move_line(
+        self, cr, uid, inv_move_line, new_line_amount, context=None
+    ):
+        if inv_move_line.account_id.id == inv_move_line.real_account_id.id:
+            line_amount = 0.00
+        else:
+            line_amount = new_line_amount
+        vals = {
+            'name': inv_move_line.name,
+            'account_id': inv_move_line.account_id.id,
+            'credit': (inv_move_line.debit and line_amount or 0.0),
+            'debit': (inv_move_line.credit and line_amount or 0.0),
+            'type': 'shadow',
+            'partner_id': (inv_move_line.partner_id and
+                           inv_move_line.partner_id.id or False)
+        }
+        if inv_move_line.tax_code_id:
+            vals[
+                'tax_code_id'
+            ] = inv_move_line.tax_code_id.id
+            if inv_move_line.tax_amount < 0:
+                vals['tax_amount'] = new_line_amount
+            else:
+                vals['tax_amount'] = -new_line_amount
+        return vals
+
+    def _prepare_shadow_move(self, cr, uid, bank_line, context=None):
+        vat_config_error = bank_line.company_id.vat_config_error
+        if not bank_line.journal_id.vat_on_payment_related_journal_id:
+            if vat_config_error == 'raise_error':
+                raise orm.except_orm(
+                    _('Error'),
+                    _("We are on a VAT on payment treatment "
+                      "but journal %s does not have a related shadow "
+                      "journal")
+                    % bank_line.journal_id.name)
+            else:
+                real_journal = bank_line.journal_id.id
+        else:
+            real_journal = (
+                bank_line.journal_id.vat_on_payment_related_journal_id.id)
+        return {
+            'journal_id': real_journal,
+            'period_id': bank_line.statement_id.period_id.id,
+            'date': bank_line.statement_id.date,
+        }
+
+    def _move_payment_lines_to_shadow_entry(
+        self, cr, uid, bank_line, shadow_move_id, context=None
+    ):
+        for line in bank_line.journal_entry_id.line_id:
+            if line.account_id.type != 'liquidity':
+                # If the line is related to write-off and user doesn't
+                # want to compute the tax including write-off,
+                # write-off move line must stay on the real move
+                line.write({'move_id': shadow_move_id}, update_check=False)
+                # this will allow user to see the real entry from
+                # invoice payment tab
+                if (line.account_id.type == 'receivable' or
+                        line.account_id.type == 'payable'):
+                    line.write({
+                        'real_payment_move_id': bank_line.journal_entry_id.id,
+                    })
+        return True
+
+    def _create_vat_on_payment_move(self, cr, uid, bank_line, mv_line_dicts,
```

Comment:
@eLBati I will do it today, but i think it's more usefull to do it directly on the odoo base, so we can use afterwards...